plutus-tx/src/PlutusTx/Blueprint/Schema.hs

```diff
@@ -0,0 +1,417 @@
+{-# LANGUAGE AllowAmbiguousTypes      #-}
+{-# LANGUAGE DataKinds                #-}
+{-# LANGUAGE DerivingStrategies       #-}
+{-# LANGUAGE LambdaCase               #-}
+{-# LANGUAGE OverloadedStrings        #-}
+{-# LANGUAGE PolyKinds                #-}
+{-# LANGUAGE StandaloneKindSignatures #-}
+{-# LANGUAGE TypeApplications         #-}
+{-# LANGUAGE TypeFamilies             #-}
+{-# LANGUAGE TypeOperators            #-}
+{-# LANGUAGE UndecidableInstances     #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+module PlutusTx.Blueprint.Schema where
+
+import Data.Aeson (ToJSON (..), (.=))
+import Data.Aeson qualified as Aeson
+import Data.ByteString (ByteString)
+import Data.ByteString.Base16 qualified as Base16
+import Data.Kind (Constraint)
+import Data.List.NonEmpty (NonEmpty)
+import Data.Maybe (catMaybes)
+import Data.Proxy (Proxy (..))
+import Data.Text (Text)
+import Data.Text.Encoding qualified as Text
+import Data.Typeable (Typeable, typeRep)
+import GHC.TypeLits qualified as GHC
+import Numeric.Natural (Natural)
+import Prelude hiding (max, min)
+
+data Schema
+  = SchemaInteger
+      -- | Title
+      (Maybe String)
```

Comment:
Suggest using a record type for each variant, and using newtypes when appropriate.